aurora/source_utils.py

```diff
@@ -266,7 +266,13 @@ def get_radial_source(namelist, rvol_grid, pro_grid, S_rates,nt, Ti_eV=None):
         Radial profile of the impurity neutral source for each time step.
     '''
     r_src = namelist['rvol_lcfs'] + namelist['source_cm_out_lcfs']
-
+    nt = S_rates.shape[1]
+    try:
+        # TODO: invert order of dimensions of Ti_eV...
+        assert S_rates.shape==Ti_eV.T.shape
```

Comment:
When I have tried to run it from IMPRAD, it has crashed exactly here. You call this function from core.py as 

```
 self.rcl_rad_prof = source_utils.get_radial_source(
                    nml_rcl_prof, # namelist specifically to obtain exp decay from wall
                    self.rvol_grid, self.pro_grid,
                    self.S_rates[:,0,:],   # 0th charge state (neutral)
                    self._Ti[[0],:])
```

why are you using in argument self._Ti[[0],:] - this is a first timeslice of Ti. 
_get_radial_source_ should be working for time-evolving Ti profiles.

Reply:
I've corrected this now. 
I've made the recycling radial profile completely time INDEPENDENT because otherwise one would need to pass yet another large array to fortran/julia. This seems particularly inconvenient because most often users (a) may not even have recycling turned on, and (b) even when they do, they may want to set their own recycling profile (now possible with this new branch). It's just a choice led by pragmatism and simplicity. Let me know if you disagree.

Reply:
I need to check if it is not causing some issues with particle conservation. If not, than it is fine.